src/cloudWatchLogs/commands/searchLogGroup.ts

```diff
@@ -73,35 +117,184 @@ async function logGroupsToArray(logGroups: AsyncIterableIterator<CloudWatchLogs.
+            {
+                label: 'Custom time range',
+                data: customRange,
+                detail: `YYYY/MM/DD-YYYY/MM/DD`,
+            },
+        ]
+
+        return prompter
+    }
+
+    private switchState(newState: 'custom-range' | 'recent-range') {
+        this.currentState = newState
+    }
+
+    public createDateBox(): InputBoxPrompter {
+        return createInputBox({
+            title: 'Enter custom date range',
+            placeholder: 'YYYY/MM/DD-YYYY/MM/DD',
+            validateInput: input => this.validateDate(input),
+        })
+    }
+
+    protected async promptUser(): Promise<PromptResult<TimeFilterResponse>> {
+        while (true) {
+            switch (this.currentState) {
+                case 'recent-range': {
+                    this.steps && this.defaultPrompter.setSteps(this.steps[0], this.steps[1])
+
+                    const resp = await this.defaultPrompter.prompt()
+                    if (resp === customRange) {
+                        this.switchState('custom-range')
+                    } else if (isValidResponse(resp)) {
+                        const [endTime, startTime] = [new Date(), new Date()]
+                        startTime.setHours(endTime.getHours() - resp)
+
+                        return { start: startTime.valueOf(), end: endTime.valueOf() }
+                    } else {
+                        return resp
+                    }
+
+                    break
+                }
+                case 'custom-range': {
+                    const resp = await this.customPrompter.prompt()
+                    if (isValidResponse(resp)) {
+                        const [startTime, endTime] = this.parseDate(resp)
+
+                        return { start: startTime.valueOf(), end: endTime.valueOf() }
+                    }
+
+                    this.switchState('recent-range')
+
+                    break
+                }
+            }
+        }
+    }
+
+    public validateDate(input: string) {
+        const parts = input.split('-')
+        const today = new Date()
+
+        if (parts.length !== 2) {
+            return 'String must include two dates seperated by `-`'
+        }
+        const [startTime, endTime] = parts
+
+        if (!Date.parse(startTime)) {
+            return 'starting time format is invalid, use YYYY/MM/DD'
+        }
+        if (!Date.parse(endTime)) {
+            return 'ending time format is valid, use YYYY/MM/DD'
+        }
+        const regEx = /^\d{4}\/\d{2}\/\d{2}$/
+        if (!startTime.match(regEx) || !endTime.match(regEx)) {
+            return 'enter date in format YYYY/MM/DD-YYYY/MM/DD'
+        }
+        if (startTime === endTime) {
+            return 'must enter two different dates for valid range'
+        }
+        if (Date.parse(startTime) > Date.parse(endTime)) {
+            return 'first date must occur before second date'
+        }
+
+        if (Date.parse(endTime) > today.valueOf()) {
```

Comment:
Should we also validate if startTime  is > today.valueOf() as well? E.g. if a user enters 2025/01/01 as the start time and 2026/01/01 as the end time

Reply:
I think that case is covered since I also check that the end time occurs after the start date. If `startTime > today.valueOf()` but it passes the previous check so we know that  `startTime < endTime` then it must also be true that `endTime > today.valueOf()` so it would fail this test. Let me know if I am missing something, but I believe it's covered.